### PR TITLE
Get a mostly-working local Trac+kenaan install

### PR DESCRIPTION
**Pull request by hawkowl**

### CODE REVIEW

File: provisioning/playbook.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+- hosts: dornkirk-staging
+  sudo: yes
+  tasks:
+    - name: Add vagrant key
+      action: authorized_key user=root key="ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkTkyrtvp9eWW6A8YVr+kz4TjGYe7gHzIw+niNltGEFHzD8+v1I2YJ6oXevct1YeS0o9HZyN1Q9qgCgzUFtdOKLv6IedplqoPkcmF0aYet2PkEDo3MlTBckFXPITAMzF8dJSIFo9D8HfdOV0IAdx4O7PtixWKn5y2hMNG0zQPyUecp4pzC6kivAIhyfHilFR61RGL+GPXQ2MWZWFYbAGjyiYJnAmCP3NOTd0jMZEnDkbUvxhMmBYSdETk1rRgm+R4LOzFUGaHqHDLKLX+FIPKcF96hrucXzcWyLbIbEgE98OHlnVYCzRdK8jlqm8tehUc9c9WhQ== vagrant insecure public key"
```
Comment (glyph) [inline provisioning/playbook.yml]: Is... is this really how you do that?

Comment (hawkowl) [inline provisioning/playbook.yml]: Not really. I should really use the current user's public key instead.

**Comment by hawkowl** [comment]
To test this:

```
vagrant up
fab config.vagrant base.bootstrap
fab config.vagrant t-web.install
fab config.vagrant t-web.makeTestTLSKeys
fab config.vagrant t-web.installSSLKeys
fab config.vagrant trac.install
fab config.vagrant trac.installTestData
fab config.vagrant amptrac.install
fab config.vagrant kenaan.install
fab config.vagrant t-web.start
fab config.vagrant trac.start
fab config.vagrant amptrac.start
fab config.vagrant kenaan.start
```

Then add `172.16.255.140 twistedmatrix.com` in your hosts file and register an account. If you make tickets, it'll go to the place in the config.py.sample.

If you review or 'fix' tickets, it will make things on the actual channels, so don't do that. :)

**Comment by adiroiban** [comment]
What is the scope of this PR?

What problems does it fix?

The title is 'Get a semi-working vagrant' ... what is that? How do you define semi-working?

Why not merge https://github.com/twisted-infra/braid/pull/64 ?

I feel that it does to many things from a single shot.

Thanks!

**Comment by adiroiban** [comment]
I was also expecting to see some updates in the README files describing how to use the stating services... or what to do with the ansible stuff.

**Comment by hawkowl** [comment]
This one incorporates #64, but had so many crazy merge conflicts that I wasn't able to get it to work.

I'll have to add some comments tomorrow. In short, this should provide enough locally working trac so that I can work on migrating the password database to a real database, and provide the groundwork for getting a whole operating Trac setup in staging.

The ansible stuff is handled by vagrant. Go `vagrant up` and it'll apply it on creation, or go `vagrant provision` to reapply it.

**Comment by froztbyte** [comment]
Why is one task called "make tls keys" and the other "install ssl keys"? Perhaps a minor bikeshed, but it'd make more sense if both match up. 

Sent from my HTC

----- Reply message -----

**Comment by adiroiban** [comment]
Please update the documentation (README) files so that when a new twisted-infra people want to start working on this they can just use the information from the README files or wiki pages and don't have to search for PR comments to know how to use the trac staging.

Thanks!

**Comment by glyph** [comment]
@froztbyte - are you just saying it should say "TLS" in both places? :)

**Comment by froztbyte** [comment]
Yes, and a reference to "THERE IS NO SSL". It was just odd to see one half having changed in a PR with the other HD untouched. 

Sent from my HTC

----- Reply message -----

**Comment by adiroiban** [comment]
This is a big change in infrastructure and it comes with little documentation or explanation about why those changes are needed.

Can you please break this into smaller patches and submit  them together with a cover letter and documentation?

Thanks!

**Pull request closed without merging**